core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -372,27 +386,34 @@ object TopicCommand extends Logging {
                 val numPartitions = topicPartitionAssignment.size
                 val replicationFactor = topicPartitionAssignment.head._2.size
                 val configsAsString = configs.map { case (k, v) => s"$k=$v" }.mkString(",")
-                val markedForDeletionString = if (markedForDeletion) "\tMarkedForDeletion:true" else ""
-                println(s"Topic:$topic\tPartitionCount:$numPartitions\tReplicationFactor:$replicationFactor\tConfigs:$configsAsString$markedForDeletionString")
+                println(s"Topic: $topic" +
+                  s"\tPartitionCount: $numPartitions" +
+                  s"\tReplicationFactor: $replicationFactor" +
+                  s"\tConfigs: $configsAsString" +
+                  (if (markedForDeletion) "\tMarkedForDeletion: true" else ""))
               }
             }
             if (describeOptions.describePartitions) {
               for ((partitionId, assignedReplicas) <- topicPartitionAssignment.toSeq.sortBy(_._1)) {
-                val leaderIsrEpoch = zkClient.getTopicPartitionState(new TopicPartition(topic, partitionId))
-                val partitionDesc = PartitionDescription(
-                  topic,
-                  partitionId,
-                  leader = if (leaderIsrEpoch.isEmpty) None else Option(leaderIsrEpoch.get.leaderAndIsr.leader),
-                  assignedReplicas,
-                  isr = if (leaderIsrEpoch.isEmpty) Seq.empty[Int] else leaderIsrEpoch.get.leaderAndIsr.isr,
-                  minIsrCount = 0,
-                  markedForDeletion,
-                  describeOptions.describeConfigs
-                )
-
-                if (describeOptions.shouldPrintTopicPartition(partitionDesc)) {
-                  printPartition(partitionDesc)
+                val tp = new TopicPartition(topic, partitionId)
+                val (leaderOpt, isr) =  zkClient.getTopicPartitionState(tp).map(_.leaderAndIsr) match {
+                  case Some(leaderAndIsr) => (leaderAndIsr.leaderOpt, leaderAndIsr.isr)
+                  case None => (None, Seq.empty[Int])
+                }
+
+                def asNode(brokerId: Int): Node = {
```

Comment:
This adapting is admittedly a little awkward, but eventually this code goes away, so I optimized for the admin client path.

Reply:
What do you mean by "eventually this goes away"?

Reply:
We are planning to deprecate and remove the --zookeeper option.